lib/spack/llnl/util/link_tree.py

```diff
@@ -112,4 +126,6 @@ def unmerge(self, dest_root, **kwargs):
             elif os.path.exists(dest):
                 if not os.path.islink(dest):
                     raise ValueError("%s is not a link tree!" % dest)
-                os.remove(dest)
+                # only remove if src actually points to dest
+                if filecmp.cmp(src, dest, shallow=True):
```

Comment:
Would it be clearer to use ```if os.path.realpath(dest) == src```? IMO it is sufficient if ```dest``` is in fact a symlink to ```src```; if you agree with me on that IMO this is more straightforward since it takes more effort to understand the implications of ```filecmp.cmp```.